includes/class-reader-activation.php

```diff
@@ -737,10 +737,11 @@ public static function set_current_reader( $user_id ) {
 	 * @param string $email        Email address.
 	 * @param string $display_name Reader display name to be used on account creation.
 	 * @param bool   $authenticate Whether to authenticate after registering. Default to true.
+	 * @param object $metadata Any metadata to pass along to the action hook.
 	 *
 	 * @return int|false|\WP_Error The created user ID in case of registration, false if the user already exists, or a WP_Error object.
 	 */
-	public static function register_reader( $email, $display_name = '', $authenticate = true ) {
+	public static function register_reader( $email, $display_name = '', $authenticate = true, $metadata = false ) {
```

Comment:
```suggestion
	public static function register_reader( $email, $display_name = '', $authenticate = true, $metadata = [] ) {
```

We can default this to an empty array, no reason to use boolean type here.